go-eicio/writer.go

```diff
@@ -15,11 +17,23 @@ func Create(filename string) (*Writer, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if strings.HasSuffix(filename, ".gz") {
+		return NewGzipWriter(file)
```

Comment:
I've addressed this by handling the deconstructor-ish calls more systematically in #7.  The user is free to directly use `NewWriter` or use the `Create`, `Open`, `Close`, `NewGzipWriter`, and `NewGzipReader` helper functions.